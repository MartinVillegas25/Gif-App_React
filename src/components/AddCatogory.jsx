import { useState } from "react"
import PropTypes from "prop-types"


export const AddCatogory = ( { onNewcategoty}) => {


    const [ inputValue, setInputValue] = useState()

    //vamos a usuar el metodo onChange para poder cambiar el valor del input

    const onInputChange = (event)=>{
            setInputValue(event.target.value)
    }

    const onSubmit = (e)=>{
            e.preventDefault();
            const inputValueTrim = inputValue || "";
            if(inputValueTrim.trim().length <= 1) return;
            setInputValue( "")
            onNewcategoty(inputValue.trim());
    }
  return (
    <form onSubmit={ onSubmit} aria-label="form">
        <input
            type="text"
            placeholder="Buscar Gif"
            value= {inputValue}
            onChange={onInputChange}
        />

    </form>
    
  )
}


AddCatogory.propTypes = {
  onNewcategoty: PropTypes.func.isRequired,
}