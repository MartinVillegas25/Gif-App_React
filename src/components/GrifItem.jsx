
import PropTypes from 'prop-types';



export const GrifItem = ( {title , url}) => {
  return (
    <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}


GrifItem.propTypes = {
  title : PropTypes.string.isRequired,
  url : PropTypes.string.isRequired
}

