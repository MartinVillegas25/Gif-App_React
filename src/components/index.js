//archivo barril

export * from './AddCatogory'
export * from './GrifGrid'
export * from './GrifItem'