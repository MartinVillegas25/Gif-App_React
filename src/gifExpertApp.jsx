import { useState } from "react"
import { AddCatogory, GrifGrid} from "./components";



export const GifExpertApp  = () => {

  //cuando tengo q almacenar algo y eso va a cambiar el estado del component, vamos a usar un hook, hay varios, nosotros vamos a usar el useState

  const [category, setCategory] = useState([]);
  const onAddCategory = (newCategory)=>{

      if(category.includes(newCategory)) return;
      // no usasmos push, asi no lo mutamos, sino q creamos un nuevo arreglo
      setCategory([newCategory, ...category]);
      
  }

  return (
    <>

    <div className="container">
        {/* titulo */}

        <h1>GIF APP</h1>

      {/* input */}
     <AddCatogory 
    //  setCategory = {setCategory}
       
    //vamos a hacerlo de otra forma, que lo unico que haga este componente es obtener el campo y que el componente padre sea el q haga el cambio 
        onNewcategoty = { (value) => onAddCategory(value)}
     />

    </div>

      {/*Listado de gifs */}
      
   
      
     
      {/* para renderizar los valores que viene de la API, usamos una expresion de JS, donde hacemos un map del arreglo q llega y retornamos los li para cada uno, es obligatorio poner una key, no importa que sea, puede ser cualquier cosa pero debe tener una key para poder renderizarlos */}
      {category.map(category => 
        (
          <GrifGrid key={category} category={category}/>
        ))
      }
  


          {/* gif item */}


      <footer className="footer">

        <p>Curso de React</p>
        <p>Martin Villegas</p>
      </footer>

    </>
  )
}
