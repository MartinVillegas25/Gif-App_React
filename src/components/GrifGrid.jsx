import {PropTypes } from 'prop-types'
import { GrifItem } from "./GrifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";




export const GrifGrid = ( {category}) => {
    

//vamos a crear un custom hook para reemplazar toto el codigo abajo

const { images , isLoading } = useFetchGifs(category) // le podemos poner cualquier nombre, solo hay q poner  use delante del nombre, norma al crear hooks

  // const [images, setImages] = useState([]);

  // const getImages = async ()=>{
  //   const newImages = await getGif(category);
  //   setImages(newImages);
    
  // }

  // //el useEffect es un hook que sirva para disparar un efecto secundario, q es un algo que you quiero dispadar cuando algo sucesa

  // useEffect(() => {
  //   getImages();
    
  // }, []) 
  
  //con las llaves vacias, le digo q solo lo renderice la primera vez
  

  
    return (
    <>
        <h3>{category}</h3>

        {
          isLoading && (<h2>Loading...</h2>)
        }

        <div className="card-grid">
          {
            images.map( (img) => (
              <GrifItem 
              key={img.id}
              {...img} // para que el gifitem reciba cada una de las propediades como items
              />
            ))

          }
        </div>
        
    
    </>
  )
}

GrifGrid.propTypes = {
  category : PropTypes.string.isRequired

}
