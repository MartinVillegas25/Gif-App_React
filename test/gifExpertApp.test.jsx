import { render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/gifExpertApp"
import { AddCatogory } from "../src/components"




jest.mock('../src/components')

describe('pruebas en gifExpertApp', () => { 

    const inputValue = 'Saitama' 

    test('debe de considir con el snapshot', () => { 

            const {container}= render(<GifExpertApp/>);
            
            expect(container).toMatchSnapshot();
        
        
     })

    //  test('onAddCategory es llamada', () => { 

    //     render(<GifExpertApp/>);
    //     const onAddCategory = jest.fn();
    //     expect(onAddCategory).toHaveBeenCalled()

    //   })


      test('el valor existe', () => { 

        render(<GifExpertApp/>);
        
        AddCatogory.mockReturnValue(
            <>
                <form>
                    <input 
                        value={inputValue}
                        
                        />
                </form>
           </> 
        )
        
        screen.debug();
        expect(inputValue).toBeTruthy(); 

      })


    
 })