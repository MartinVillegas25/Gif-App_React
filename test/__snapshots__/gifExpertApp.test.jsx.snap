// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en gifExpertApp debe de considir con el snapshot 1`] = `
<div>
  <h1>
    GIF APP
  </h1>
</div>
`;
